plugin/clone/src/clone_client.cc

```diff
@@ -1722,6 +1722,15 @@ int Client::set_descriptor(const uchar *buffer, size_t length) {
   /* Reset buffers */
   aux_conn->reset();
 
+  // If the error happened in any other storage engine than InnoDB, notify
+  // InnoDB too to break any waits there
+  if (loc_index != 0) {
+    aux_conn->m_error = err;
```

Comment:
Maybe assert `m_share->m_storage_vec[0]->m_hton.db_type == DB_TYPE_ROCKSDB`?

Reply:
This work around is probably good enough. It feels like the Ha_clone_cbk interface should be extended to include a way to notify the clone plugin of an error.

One of your original changes was to have rocksdb_clone_ack() return the error code. This may result in an extra status being sent from the donor to the client, but it looks like it would also trigger cleanup on the donor. Would that be a better fix?

Reply:
> This work around is probably good enough. It feels like the Ha_clone_cbk interface should be extended to include a way to notify the clone plugin of an error.

Agreed.
 
> One of your original changes was to have rocksdb_clone_ack() return the error code. This may result in an extra status being sent from the donor to the client, but it looks like it would also trigger cleanup on the donor. Would that be a better fix?

That happened to be the implementation at the time issue was discovered (i.e. the SST COPY was introduced, the metadata send was put in the right place, and the testcase started failing). I believe a `COM_ACK` failure results in cleanup being triggered on the donor for that one thread only, with the other threads relying on error being passed through the SE layer. I will confirm tomorrow my morning

Reply:
> > One of your original changes was to have rocksdb_clone_ack() return the error code. This may result in an extra status being sent from the donor to the client, but it looks like it would also trigger cleanup on the donor. Would that be a better fix?
> 
> That happened to be the implementation at the time issue was discovered (i.e. the SST COPY was introduced, the metadata send was put in the right place, and the testcase started failing). I believe a `COM_ACK` failure results in cleanup being triggered on the donor for that one thread only, with the other threads relying on error being passed through the SE layer. I will confirm tomorrow my morning

The above description was not correct in how `COM_ACK` failure is handled,although the end result it the same). If the donor SE returns an error on COM_ACK, the donor sends `COM_ERROR` to the client, which then waits for donor to send something else, which never happens (hence the original "protocol desync" description of the issue)